lib/route.js

```diff
@@ -227,9 +227,7 @@ function buildRouting (options) {
         }
       }
 
-      const config = opts.config || {}
-      config.url = url
-      config.method = opts.method
+      const config = Object.assign({ url, method: opts.method }, opts.config)
```

Comment:
This is a bad bug that's present on master as well.

Reply:
Nice finding!

Reply:
Fixed in master with #2827